### ISSUE TITLE
BUG: Certain SARIMAX models cannot be fitted for data with nobs=13

### ISSUE THREAD

**Issue by Mastercol01** [statsmodels/statsmodels#7836]
I have the following pandas data-frame:

```
	ds	y
0	2020-01-31	-10
1	2020-02-29	-15
2	2020-03-31	-293
3	2020-04-30	-75
4	2020-05-31	-187
5	2020-06-30	-199
6	2020-07-31	-359
7	2020-08-31	-136
8	2020-09-30	-344
9	2020-10-31	-56
10	2020-11-30	-206
11	2020-12-31	-20
12	2021-01-31	-91
13	2021-02-28	-108
14	2021-03-31	100
15	2021-04-30	91
16	2021-05-31	17
17	2021-06-30	-117
18	2021-07-31	-60
19	2021-08-31	-87
20	2021-09-30	-20
```
And have realized that for certain subsets of said data-frame, it is not always possible to fit the SARIMAX model:
`(p=2,d=3,q=2)(P=1,D=3,Q=2,m=3)`. In essence:

**Comment by bashtage** [comment]
Your AR polynomial is too big for 13 observations.  IMO it should also error for < 13.  The maximum AR Lag is 5 (from the non seasonal part) * 12 (from the seasonal part) = 60.  This model makes little sense, and any estimate isn't going to be sensible. 

Also, trend=c makes no sense with the differencing you are doing. 

Is the statsmodels 0.13? The trend should be raising in 0.13.

**Comment by Mastercol01** [comment]
Yes, the version is statsmodels 0.13. What would be a better-suited trend term?. Regarding your observation about the AR component being too large, you're right now that I actually analyse it. In total there d+m*D differences, which in this case would mean 12 data-points lost from supposedly 13 nobs... And aside from that I'm requesting to look back up to 3 lags.... Yeah, that's my bad. However, I then agree that it should also raise an error when `nobs<13`, which currently it does not. And I also think a more descriptive error would be in order.

**Comment by bashtage** [comment]
You need trend terms that won't be differenced away.  If you have one difference, you should use trend ="t" or [1].  If you have 6 differences, you would need trend = [6] to actually have a terms that would not be differenced away.

Both of our calculations are wrong.  On one hand, there are differences. But there are also the AR terms.  You have 2+3 non seasonal lags + (1+3)*3 seasonal lags = a maximum lag length of 17 in your model.  This means that you don't even have a single cycle as long as the maximum lag length in your model.
